Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/Shadow/DirectionalLightShadowCalculator.azsli

```diff
@@ -84,20 +83,38 @@ class DirectionalShadowCalculator
     bool m_blendBetweenCascadesEnable;
 
     int m_lightIndex;
-    int m_cascadeIndex;
+    int m_minCascade;
+    int m_maxCascade;
     int m_cascadeCount;
     ShadowFilterMethod m_filterMode;
 
     Texture2DArray<float> m_directionalLightShadowmap;   
     Texture2DArray<float> m_directionalLightExponentialShadowmap;        
 };
 
-bool DirectionalShadowCalculator::IsShadowCoordInCascade(float3 shadowCoord, uint shadowMapSize)
+void DirectionalShadowCalculator::GetShadowCascadeRange(float3 shadowCoords[ViewSrg::MaxCascadeCount], int cascadeCount, uint shadowmapSize, out int minShadowCascade, out int maxShadowCascade)
 {
-    return (shadowCoord.x >= 0. && shadowCoord.x * shadowMapSize < shadowMapSize - PIXEL_MARGIN &&
-            shadowCoord.y >= 0. && shadowCoord.y * shadowMapSize < shadowMapSize - PIXEL_MARGIN &&
-            shadowCoord.z < (1. - DEPTH_MARGIN));       // Test the case where the shadowed pixel is nearer than the maximal depth
-                                                        // of the cascade.  Otherwise, escalate the next cascade.
+    float xyMax = shadowmapSize - PIXEL_MARGIN;
+    float xyMin = PIXEL_MARGIN;
+    minShadowCascade = -1;
+    maxShadowCascade = -1;
+
+    for (int index = 0; index < m_cascadeCount; ++index)
+    {
+        float3 shadowCoord = shadowCoords[index];
+        shadowCoord.xy *= shadowmapSize;
+        if (all(shadowCoord.xy >= xyMin) && all(shadowCoord.xy <= xyMax))
```

Comment:
I can see from the calling code that in the case where this never becomes true and you're left with minShadowCascade and maxShadowCascade both = -1, it is treated as lit and you never actually read from the cascades. That's not obvious from within this function. I think Sid's suggestion of clarifying what is being checked in this if statement, plus a comment where the min/max are initialized to -1 (or maybe a comment above the function declaration itself, explaining the output of the function) will help that.

Reply:
Adding a comment, I'll also move the if (m_maxCascade < 0) check up higher in `GetVisibility()` so it can early out sooner.